pkg/app/master/command/cliparamsfile.go

```diff
@@ -0,0 +1,148 @@
+package command
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/mintoolkit/mint/pkg/app"
+	"github.com/urfave/cli/v2"
+)
+
+func ParseParamsFromFile(ctx *cli.Context, xc *app.ExecutionContext, filePath string, validFlags []cli.Flag) {
+	params, err := parseJsonFileParams(filePath)
+
+	if err != nil {
+		xc.Out.Error("params.file.error", err.Error())
+		xc.Out.State("exited",
+			ovars{
+				"exit.code": -1,
+			})
+		xc.Exit(-1)
+	}
+
+	err = setFileParams(ctx, xc, params, validFlags)
+
+	if err != nil {
+		xc.Out.Error("params.file.param.value.error", err.Error())
+		xc.Out.State("exited",
+			ovars{
+				"exit.code": -1,
+			})
+		xc.Exit(-1)
+	}
+
+}
+
+func validFlagsMap(validFlags []cli.Flag) (map[string]cli.Flag, error) {
+	validFlagsMap := make(map[string]cli.Flag)
+
+	for _, flag := range validFlags {
+		if len(flag.Names()) == 0 {
+			return nil, errors.New("flag names has no values")
+		}
+
+		validFlagsMap[flag.Names()[0]] = flag
+	}
+
+	return validFlagsMap, nil
+}
+
+func flagSetString(flag cli.Flag, paramValue interface{}) (string, error) {
+	paramValueType := reflect.TypeOf(paramValue).Kind()
```

Comment:
How about accomplishing the same without using reflection, at least, not directly? A type switch on paramValue, for example, which could be combined either. with a set of custom consts for the type names or with the %T format specifier.

Reply:
Yep sure - just commited the to use a type switch without any reflect - still learning go didn't know reflect it was unable to detect errors during code compilation :/

Let me know if this is not what you expected.